domain/application/service/application.go

```diff
@@ -353,6 +358,17 @@ func (s *ApplicationService) AddUnits(ctx context.Context, name string, units ..
 	return errors.Annotatef(err, "adding units to application %q", name)
 }
 
+// GetUnitUUID returns the UUID for the named unit
```

Comment:
Is there a UnitNotFound error as part of this interface? If so it should be documented here.

Reply:
Yep it is, you're right it should be.

Added